main.c

```diff
@@ -101,6 +101,7 @@ void main(void){
     vector_init(&tests_to_run);
     load_conf_file("D:\\config.txt");
     open_output_file("D:\\kernel_tests.log");
+    srand(NULL);
```

Comment:
`srand` takes an `unsigned int`, not a pointer.

Ideally this seed would be chosen by the config but defaults to something known (like `0`).

It should also be considered to reseed before every single test, so that if new tests are added (after seed, before old tests), it won't affect the old test results (which would now get a new random stream if then new test consumed random bits).